### PR TITLE
Replace libtest/lib.rs:FnBox with std::boxed::FnBox.

### PR DESCRIPTION
**Pull request by frewsxcv**
Fixes https://github.com/rust-lang/rust/issues/41810.

### CODE REVIEW

File: src/libtest/lib.rs [hunk 1]
```diff
@@ -1720,7 +1710,7 @@ mod tests {
                 should_panic: ShouldPanic::No,
                 allow_fail: false,
             },
-            testfn: DynTestFn(Box::new(move |()| f())),
+            testfn: DynTestFn(Box::new(move || f())),
```
Comment (kennytm) [inline src/libtest/lib.rs]: These can simply be `Box::new(f)` I think.

Comment (frewsxcv) [reply src/libtest/lib.rs]: addressed in the latest force push

**Comment by frewsxcv** [comment]
assigning to someone random from the compiler team (unless there's a better team to review this)

r? @estebank

**Pull request merged**